kpter/spec/models/community_spec.rb

```diff
@@ -3,63 +3,175 @@
+      expect(@closed_tcards).to be_asc -> (t_card) { t_card.deadline }
     end
   end
 
-  describe "コミュニティ内にユーザーを参加させる" do
-    context "参加していない場合" do
-      it "コミュニティ・ユーザー関連テーブルに登録したコミュニティIDが引数に指定したIDと一致すること" do
-      end
-      it "コミュニティ・ユーザー関連テーブルに登録したユーザーIDが引数に指定したIDと一致すること" do
-      end
+  describe "コミュニティ内からユーザーを脱退させる" do
+    before :all do
+      @community = Community.create(name: "community name")
+      @user = FactoryGirl.create(:user)
+      # コミュニティに参加させる
+      @community_user = CommunityUser.create(community_id: @community.id, user_id: @user.id)
+      # 脱退する
+      @community.withdraw(@user)
+      @community_user = CommunityUser.where(community_id: @community.id, user_id: @user.id)
+
     end
-    context "既に参加している場合" do
-      it "エラーメッセージが設定されること" do
-      end
-      it "コミュニティ・ユーザー関連テーブルに登録されていないこと" do
+    context "コミュニティに参加しているユーザーが退会する場合" do
+      it "コミュニティID, ユーザーIDに紐づくコミュニティ・ユーザー関連テーブルが削除されること" do
+        expect(@community_user.empty?).to be true
       end
     end
-  end
-
-  describe "コミュニティ内からユーザーを除名する" do
-    it "コミュニティID, ユーザーIDに紐づくコミュニティ・ユーザー関連テーブルが削除されること" do
+    context "コミュニティに参加していないユーザーが退会する場合" do
+      it "エラーが発生しないこと" do
+        @community.withdraw(@user)
+      end
     end
-    # 既に除名済みかどうかのチェックは不要
   end
 
   describe "コミュニティ内のユーザーを全員取得する" do
-    it "コミュニティIDに紐づくコミュニティ・ユーザー関連テーブルが取得できること" do
+    before :all do
+      @community = Community.create(name: "community name")
+      # user_1, 2, 3, 4, 5があらわれた
+      @user_1 = FactoryGirl.create(:user)
+      @user_2 = FactoryGirl.create(:user)
+      @user_3 = FactoryGirl.create(:user)
+      @user_4 = FactoryGirl.create(:user)
+      @user_5 = FactoryGirl.create(:user)
+
+      # user_4, 2, 1, 5, 3の順でコミュニティに参加した
+      CommunityUser.create(community_id: @community.id, user_id: @user_4.id)
+      CommunityUser.create(community_id: @community.id, user_id: @user_2.id)
+      CommunityUser.create(community_id: @community.id, user_id: @user_1.id)
+      CommunityUser.create(community_id: @community.id, user_id: @user_5.id)
+      CommunityUser.create(community_id: @community.id, user_id: @user_3.id)
+
+      @joining_users = @community.find_users
+
+    end
+    it "コミュニティ内のユーザーが全員取得できていること" do
+      expect(@joining_users.count()).to eq(5)
+      expect(@joining_users.find(id: @user_1.id)).to be_present
+      expect(@joining_users.find(id: @user_2.id)).to be_present
+      expect(@joining_users.find(id: @user_3.id)).to be_present
+      expect(@joining_users.find(id: @user_4.id)).to be_present
+      expect(@joining_users.find(id: @user_5.id)).to be_present
     end
-    it "コミュニティ・ユーザー関連テーブルのユーザーIDに紐づくユーザーが取得できること" do
+    it "並び順が参加した順(IDの昇順)になっている" do
+      expect(@joining_users[0].id).to eq(@user_4.id)
+      expect(@joining_users[1].id).to eq(@user_2.id)
+      expect(@joining_users[2].id).to eq(@user_1.id)
+      expect(@joining_users[3].id).to eq(@user_5.id)
+      expect(@joining_users[4].id).to eq(@user_3.id)
```

Comment:
expext(@joining_users.map { |u| u.id }.to eq([@user_4.id, @user_2.id, @user_1.id, @user_5.id, @user_3.id]))